lib/any.ml

```diff
@@ -0,0 +1,7 @@
+module type Any = sig
+  type t
+end
+
+implicit module Any_Int = struct type t = int end
+implicit module Any_String = struct type t = string end
+implicit module Any_Pair { A : Any } {B : Any } = struct type t = A.t * B.t end
```

Comment:
@dvlasits Generics could help with this?

Reply:
What's the goal with this?

Reply:
`Any` is a module type for a module which does nothing except contain a type - so that types can be passed around implicitly to other modules. For example this is needed in https://github.com/modular-implicits/imp/blob/e11a836c67c73d9f5fe83c22a683d1aac8a62e64/lib/control.ml#L201-L205

Reply:
This may cause issues down the line. Any module that has a `t` member (e.g. anything that matches `Num` or `Show` or `Monoid`) will match `Any`, so it's very likely that there'll be multiple matching instances whenever `Any` is instantiated.

There's been discussion of a language design that avoids the need for `Any` implicit arguments, by allowing modules with type parameters, but it's not implemented at the moment.